utils/retry/retry.go

```diff
@@ -0,0 +1,62 @@
+package retry
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog"
+)
+
+// BackoffExponential makes maxRetry attempts to executed function f. After the nth attempt,
```

Comment:
could you update this godoc to correspond to the new method. For example, we now return a bool rather than an error, let's describe what that return value means and under what conditions we return. Likewise for `WithTimeout`